include/Grid.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#pragma once
+
+#include <SDL.h>
+#include "Color.h"
+#include "Rect.h"
+#include "Object.h"
+#include <vector>
+#include <algorithm>
+#include <utility>
+
+namespace ccm
+{
+	class Grid
+	{
+	private:
+		class Tile
+		{
+		public:
+			Tile();
+			Tile(const Rect& pos);
+			Tile(int x, int y, int w, int h);
+
+			Object draw() const;
+
+			void fill();
+			void empty();
+
+		private:
+			Rect m_position;
+			bool m_isFilled;
+		};
```

Comment:
I would personally like to see this declared in its own header+source files.

Reply:
Additionally, `m_position` suggests only an X,Y coordinate, while in reality it is more of a position+size

Reply:
Same goes for the `m_position` in Grid actually

Reply:
You mean as a standalone class. i.e not nested inside grid?

Reply:
Exactly! It makes the files a lot cleaner I think